gameEditor/Node.js

```diff
@@ -0,0 +1,20 @@
+export default class Node {
+    text = '';
+    num;
+    children = [];
+    parents = [];
+    constructor(text, num) {
+        this.text = text;
+        this.num = num;
+    }
+
+    addChild(newNode, prompt) {
+        this.children.push({prompt: prompt, node: newNode});
```

Comment:
nitpick: JS support setting props by name for a local that has a value so you can do something like:

```
addChild(node, prompt) {
    this.children.push({ node, prompt});
    ...
```